src/Microsoft.SqlTools.ServiceLayer/Admin/AdminService.cs

```diff
@@ -175,25 +175,22 @@ internal static DatabaseInfo GetDatabaseInfo(ConnectionInfo connInfo)
         internal static DatabaseTaskHelper CreateDatabaseTaskHelper(ConnectionInfo connInfo, bool databaseExists = false)
         {
             XmlDocument xmlDoc = CreateDataContainerDocument(connInfo, databaseExists);
-            char[] passwordArray = connInfo.ConnectionDetails.Password.ToCharArray();
             CDataContainer dataContainer;
 
             // check if the connection is using SQL Auth or Integrated Auth
             if (string.Equals(connInfo.ConnectionDetails.AuthenticationType, "SqlLogin", StringComparison.OrdinalIgnoreCase))
             {
-                unsafe
-                {
-                    fixed (char* passwordPtr = passwordArray)
-                    {
-                        dataContainer = new CDataContainer(
-                            CDataContainer.ServerType.SQL,
-                            connInfo.ConnectionDetails.ServerName,
-                            false,
-                            connInfo.ConnectionDetails.UserName,
-                            new System.Security.SecureString(passwordPtr, passwordArray.Length),
-                            xmlDoc.InnerXml);
-                    }
+                var passwordSecureString = new System.Security.SecureString();
+                foreach (char c in connInfo.ConnectionDetails.Password) {
+                    passwordSecureString.AppendChar(c);
                 }
+                dataContainer = new CDataContainer(
+                    CDataContainer.ServerType.SQL,
+                    connInfo.ConnectionDetails.ServerName,
+                    false,
+                    connInfo.ConnectionDetails.UserName,
```

Comment:
This code ends up passing in a blank SecureString, since it will never enter the foreach loop above if the password is empty.

We might be able to get away with passing null instead of creating a SecureString for empty passwords, but I don't know what assumptions the code further down the stack makes about whether it is null. I think it's safer to just use a blank SecureString, which clearly represents an empty string.